src/qtgui/dockfft.cpp

```diff
@@ -365,24 +376,32 @@ void DockFft::readSettings(QSettings *settings)
         ui->fftAvgSlider->setValue(intval);
 
     // Plot scale and denominator
-    int plot_scale = settings->value("plot_y_unit", 0).toInt(&conv_ok);
-    if (!conv_ok)
-        plot_scale = DEFAULT_PLOT_SCALE;
-    ui->plotScaleBox->setCurrentIndex(plot_scale);
-    int plot_per = settings->value("plot_x_unit", 0).toInt(&conv_ok);
-    if (!conv_ok)
-        plot_per = DEFAULT_PLOT_PER;
-    ui->plotPerBox->setCurrentIndex(plot_per);
+    strval = settings->value("plot_y_unit", "dbfs").toString();
+    if      (strval == "dbv") intval = 1;
+    else if (strval == "dbm") intval = 2;
+    else                      intval = 0;  // "dbfs", default
+    ui->plotScaleBox->setCurrentIndex(intval);
+    int plot_scale = intval;
+
+    strval = settings->value("plot_x_unit", "rbw").toString();
+    if      (strval == "hz") intval = 1;
+    else                     intval = 0;  // "rbw", default
+    ui->plotPerBox->setCurrentIndex(intval);
     // Trigger additional required logic for plot_scale and plot_per
     on_plotScaleBox_currentIndexChanged(plot_scale);
 
-    intval = settings->value("plot_mode", 0).toInt(&conv_ok);
-    if (conv_ok)
-        ui->plotModeBox->setCurrentIndex(intval);
-
-    intval = settings->value("waterfall_mode", 0).toInt(&conv_ok);
-    if (conv_ok)
-        ui->wfModeBox->setCurrentIndex(intval);
+    strval = settings->value("plot_mode", "max").toString();
+    if      (strval == "avg")  intval = 1;
+    else if (strval == "fill") intval = 2;
+    else if (strval == "hist") intval = 3;
+    else                       intval = 0;  // "max", default
+    ui->plotModeBox->setCurrentIndex(intval);
+
+    strval = settings->value("waterfall_mode", "max").toString();
+    if      (strval == "avg")  intval = 1;
+    else if (strval == "sync") intval = 2;
+    else                       intval = 0;  // "max", default
+    ui->wfModeBox->setCurrentIndex(intval);
```

Comment:
There's a duplicate `waterfall_mode` decoder down on lines 464-468 that still reads an integer. I presume that was a copy & paste error, and should be deleted.